app/src/main/java/edu/uc/group/rankine/ui/createRank/CreateRankSetViewModel.kt

```diff
@@ -83,10 +83,11 @@ class CreateRankSetViewModel(activity: Activity) : ViewModel() {
 
             if (snapshot != null) {
                 val allObjectSets = ArrayList<ObjectSet>()
-                val documents = snapshot.documents
+                val documents = snapshot.documents.toSet()
                 documents.forEach {
                     val objectSet = it.toObject(ObjectSet::class.java)
                     if (objectSet != null) {
+                        allObjectSets.distinct()
```

Comment:
This returns a distinct set, but you're not doing anything with the value returned.  It does not modify the set named allObjectsSet.  Probably best to remove this.